pkg/controller/interceptor/interceptors.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package interceptor offers reconciler interceptors
+package interceptor
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+)
+
+// LoggingInterceptor wraps a Reconciler and logs messages
+// before and after the reconciliation,
+// reporting if the operation was successful together with its duration.
+func LoggingInterceptor(log logging.Logger, r reconcile.Reconciler) reconcile.Func {
+	return func(ctx context.Context, req reconcile.Request) (res reconcile.Result, err error) {
+		l := log.WithValues("request", req)
+		l.Info("Reconciling")
+		start := time.Now()
+		defer func() {
+			d := time.Since(start).String()
+			if err == nil {
+				l.Info("Reconciliation done", "result", res, "duration", d)
+			} else {
+				l.Info("Reconciliation failed", "error", err, "duration", d)
+			}
+		}()
+		return r.Reconcile(ctx, req)
+	}
+}
```

Comment:
We currently don't log much at `Info` level, so I think it would make more sense to at least lower the level here to `Debug` or if you feel this needs to be at `Info` level, please start a conversation to change the project's approach to logging (if there is no dedicated issue already) and move this to a dedicated PR, so that it doesn't hold up the rest of the PR 🙏 WDYT @pedjak?

Reply:
and please split into two commits. This is not one change.

Reply:
> We currently don't log much at Info level, 

True,  we do it in a handful of places: https://github.com/search?q=repo%3Acrossplane%2Fcrossplane%20%22.Info(%22&type=code

Given that this PR is about lowering the number of reconcile rounds, I thought it would be good also to make that trackable even if debug logs are not turned on, hence this change here.

Having single info line for each reconciliation improves UX, i.e. user get a notion what is going on, i.e. what gets reconciled. Today this mostly happens silently and it is super hard to diagnose whether it happens and why.

Reply:
We do expose prometheus metrics with similar, although less granular, [metrics](https://book.kubebuilder.io/reference/metrics-reference). But, I do agree we could benefit from being more verbose, I just feel that's a major policy change that should be discussed more broadly.

Reply:
`LoggingInterceptor` moved into the separate PR: #5143.